src/Provider/ActiveMQ.php

```diff
@@ -0,0 +1,190 @@
+        $this->stomp->unsubscribe($subscriptionId);
+
+        if (false === $frame) {
+            // No queue existing
+            return 0;
+        }
+
+        $statistics = new \SimpleXMLElement($frame->body);
+        if (false === $statistics) {
+            throw new \RuntimeException("Unable to decode statistics frame body");
+        }
+
+        $elements = $statistics->xpath("/map/entry[string='size'][1]");
+        $element  = array_shift($elements);
+
+        return intval($element->long);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createQueue($queueName, array $queueOptions = [])
+    {
+        return new Queue($queueName, $this);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteQueue($queueName)
+    {
+        // Queue is deleted automatically when no more messages exists in queue
+        while ($this->get($queueName, 1) !== null) {
+            // Do nothing
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($queueName, $timeout = null)
+    {
+        $this->client->getConnection()->setReadTimeout($timeout ?: 10);
+
+        $queueName = $this->getQueueName($queueName);
+
+        $subId = $this->stomp->subscribe($queueName);
+        $frame = $this->stomp->read();
+        $this->stomp->unsubscribe($subId);
+
+        return $frame ? unserialize($frame->body) : null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function multiPut($queueName, array $workloads)
+    {
+        $this->stomp->begin();
+        foreach ($workloads as $workload) {
+            $this->put($queueName, $workload);
+        }
+        $this->stomp->commit();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function put($queueName, $workload)
+    {
+        $this->stomp->send(
+            $this->getQueueName($queueName),
+            new Message(serialize($workload), ['persistent' => 'true'])
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function queueExists($queueName)
+    {
+        $tempQueueName = sprintf("/temp-queue/%s", uniqid());
+
+        $this->client->sendFrame(new Message(
+            "",
+            [
+                "destination" => $tempQueueName,
+            ]
+        ), true);
+
+        $this->client->sendFrame(new Message(
+            "",
+            [
+                "reply-to"    => $tempQueueName,
+                "destination" => "ActiveMQ.Statistics.Destination.".$queueName,
+            ]
+        ), true);
+
+        $subscriptionId = $this->stomp->subscribe($tempQueueName);
+
+        // Ignore first frame
+        $this->stomp->read();
+        $frame = $this->stomp->read();
+        $this->stomp->unsubscribe($subscriptionId);
+
+        return false !== $frame;
+    }
+
+    /**
+     * @param string $queueName
+     *
+     * @return string
+     */
+    private function getQueueName($queueName)
+    {
+        return sprintf("/queue/%s", $queueName);
```

Comment:
Do it have any limitation on the queue name ? special characters ? space ?

Reply:
No limitation,  i just added some tests